tools/analyze_functions.py

```diff
@@ -0,0 +1,134 @@
+    print(f"Total functions without docstrings: {count_functions(functions_info, 'functions_without_docstrings')}")
+
+def compare(source_file, target_file):
+    print(f"Comparing {source_file} with {target_file}")
+    source_info = yaml.safe_load(open(source_file))
+    target_info = yaml.safe_load(open(target_file))
+
+    source_type_hints = count_functions(source_info, 'functions_without_type_hints')
+    target_type_hints = count_functions(target_info, 'functions_without_type_hints')
+
+    source_docstrings = count_functions(source_info, 'functions_without_docstrings')
+    target_docstrings = count_functions(target_info, 'functions_without_docstrings')
+
+    print(f'Source branch - functions without type hints: {source_type_hints}, without docstrings: {source_docstrings}')
+    print(f'Target branch - functions without type hints: {target_type_hints}, without docstrings: {target_docstrings}')
+    
+    errors = []
+    new_functions_without_type_hints = []
+    new_functions_without_docstrings = []
+
+    for file, data in source_info.items():
+        source_functions_without_type_hints = set(data['functions_without_type_hints'])
+        target_functions_without_type_hints = set(target_info.get(file, {}).get('functions_without_type_hints', []))
+        new_type_hints = source_functions_without_type_hints - target_functions_without_type_hints
+        if new_type_hints:
+            new_functions_without_type_hints.extend([f"{file}: {func}" for func in new_type_hints])
+
+        source_functions_without_docstrings = set(data['functions_without_docstrings'])
+        target_functions_without_docstrings = set(target_info.get(file, {}).get('functions_without_docstrings', []))
+        new_docstrings = source_functions_without_docstrings - target_functions_without_docstrings
+        if new_docstrings:
+            new_functions_without_docstrings.extend([f"{file}: {func}" for func in new_docstrings])
+
+    if new_functions_without_type_hints:
+        errors.append('Error: Number of functions without type hints increased')
+        print("New functions without type hints:")
+        for func in new_functions_without_type_hints:
+            print(f"  {func}")
+
+    if new_functions_without_docstrings:
+        errors.append('Error: Number of functions without docstrings increased')
+        print("New functions without docstrings:")
+        for func in new_functions_without_docstrings:
+            print(f"  {func}")
+
+    if errors:
+        for error in errors:
+            print(error)
+        sys.exit(1)
+    else:
+        print("No new issues found.")
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Analyze Python functions in directories or compare results.')
+    subparsers = parser.add_subparsers(dest='command')
+
+    analyze_parser = subparsers.add_parser('analyze', help='Analyze the functions in directories.')
```

Comment:
split this into two different commands: one for docstrings and one for type hints. this will then make it so that two yml files are generated, one for each. so changes will need to be made to github action as well.